core/src/data-generator/batch-replayer.ts

```diff
@@ -99,12 +173,102 @@ export class BatchReplayer extends Construct {
 
   /**
    * Maximum file size for each output file. If the output batch file is,
-   * larger than that, it will be splitted into multiple files that fit this size.
+   * larger than that, it will be split into multiple files that fit this size.
    *
    * Default to 100MB (max value)
    */
   public readonly outputFileMaxSizeInBytes?: number;
 
+  /**
+   * The DynamoDB table where the BatchReplayer writes data.
+   */
+  public readonly ddbTable?: ITable;
+
+  /**
+   * Redshift table where the BatchReplayer writes data.
+   */
+  public readonly redshiftTableName?: string;
+
+  /**
+   * Glue catalog connection name for Redshift cluster.
+   */
+  public readonly redshiftConnection?: string;
+
+  /**
+   * Schema name for the Redshift table.
+   */
+  public readonly redshiftSchema?: string;
+
+  /**
+   * RDS MySQL Aurora cluster where the BatchReplayer writes data.
+   */
+  public readonly auroraMysqlTableName?: string;
+
+  /**
+   * Glue catalog connection name for Aurora MySQL cluster.
+   */
+  public readonly auroraMysqlConnection?: string;
+
+  /**
+   * Schema name for the Aurora MySQL table.
+   */
+  public readonly auroraMysqlSchema?: string;
+
+  /**
+   * RDS PostgreSQL Aurora cluster where the BatchReplayer writes data.
+   */
+  public readonly auroraPostgresTableName?: string;
+
+  /**
+   * Glue catalog connection name for Aurora PostgreSQL cluster.
+   */
+  public readonly auroraPostgresConnection?: string;
+
+  /**
+   * Schema name for the Aurora PostgreSQL table.
+   */
+  public readonly auroraPostgresSchema?: string;
+
+  /**
+   * RDS MySQL database instance where the BatchReplayer writes data.
+   */
+  public readonly mysqlTableName?: string;
+
+  /**
+   * Glue catalog connection name for the MySQL database.
+   */
+  public readonly mysqlConnection?: string;
+
+  /**
+   * Schema name for the RDS MySQL table.
+   */
+  public readonly mysqlSchema?: string;
+
+  /**
+   * RDS PostgreSQL database instance where the BatchReplayer writes data.
+   */
+  public readonly postgresTableName?: string;
+
+  /**
+   * Glue catalog connection name for the PostgreSQL database.
+   */
+  public readonly postgresConnection?: string;
+
+  /**
+   * Schema name for the RDS PostgreSQL table.
+   */
+  public readonly postgresSchema?: string;
+
+  /**
+   * The VPC for the database instance(s) and/or cluster(s).
+   */
+  public readonly databaseVpc?: IVpc;
+
+  /**
+   * The security group of the database VPC. This needs to be added to the writeInBatchFn Lambda.
+   */
+  public readonly databaseVpcSG?: ISecurityGroup;
```

Comment:
create a dedicated security group for the BatchReplayer instead of reusing the DB sec group